hydra-cluster/src/CardanoNode.hs

```diff
@@ -135,21 +147,43 @@ withCardanoNodeDevnet ::
   (RunningNode -> IO a) ->
   IO a
 withCardanoNodeDevnet tracer stateDirectory action = do
-  args <- setupCardanoDevnet stateDirectory
-  withCardanoNode tracer stateDirectory args action
+  maybeRunningNode <- findRunningCardanoNode stateDirectory
+  case maybeRunningNode of
+    Nothing -> do
+      args <- setupCardanoDevnet stateDirectory
+      withCardanoNode tracer stateDirectory args action
+    Just runningNode -> action runningNode
+
+-- | Run a cardano-node as normal network participant on a known network,
+-- but only if there is no instance already running.
```

Comment:
Should: fix this haddock

It is always trying to run a cardano-node on known network.